src/commonTest/kotlin/org/cru/godtools/tool/model/MultiselectTest.kt

```diff
@@ -0,0 +1,171 @@
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import org.cru.godtools.tool.FEATURE_MULTISELECT
+import org.cru.godtools.tool.ParserConfig
+import org.cru.godtools.tool.internal.AndroidJUnit4
+import org.cru.godtools.tool.internal.RunOnAndroidWith
+import org.cru.godtools.tool.internal.UsesResources
+import org.cru.godtools.tool.internal.receive
+import org.cru.godtools.tool.internal.runBlockingTest
+import org.cru.godtools.tool.state.State
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertIs
+import kotlin.test.assertTrue
+
+@RunOnAndroidWith(AndroidJUnit4::class)
+class MultiselectTest : UsesResources() {
+    private val state by lazy { State() }
+
+    @Test
+    fun testParseMultiselect() = runBlockingTest {
+        val multiselect = Multiselect(Manifest(), getTestXmlParser("multiselect.xml"))
+        assertEquals("quiz1", multiselect.stateName)
+        assertEquals(2, multiselect.selectionLimit)
+        assertEquals(3, multiselect.options.size)
+        with(multiselect.options[1]) {
+            assertEquals("answer2", value)
+            assertEquals(1, content.size)
+            with(assertIs<Text>(content.single())) {
+                assertEquals("Answer 2", text)
+            }
+        }
+    }
+
+    @Test
+    fun testParseMultiselectDefaults() = runBlockingTest {
+        val multiselect = Multiselect(Manifest(), getTestXmlParser("multiselect_defaults.xml"))
+        assertEquals("", multiselect.stateName)
+        assertEquals(1, multiselect.selectionLimit)
+        assertEquals(1, multiselect.options.size)
+        with(multiselect.options.single()) {
+            assertEquals("valueAttr", value)
+            assertTrue(content.isEmpty())
+        }
+    }
+
+    @Test
+    fun testIsIgnored() {
+        val multiselect = Multiselect()
+
+        ParserConfig.supportedFeatures = setOf(FEATURE_MULTISELECT)
+        assertFalse(multiselect.isIgnored)
+
+        ParserConfig.supportedFeatures = emptySet()
+        assertTrue(multiselect.isIgnored)
+    }
+
+    @Test
+    fun testOptionIsSelected() {
+        val multiselect = Multiselect { it.options(2) }
+
+        multiselect.options[0].toggleSelected(state)
+        assertTrue(multiselect.options[0].isSelected(state))
+        assertFalse(multiselect.options[1].isSelected(state))
+    }
+
+    @Test
+    @OptIn(ExperimentalCoroutinesApi::class)
+    fun testOptionIsSelectedFlow() = runBlockingTest {
+        val multiselect = Multiselect { it.options(2) }
+
+        val flowOutput = Channel<Boolean>(1)
+        val flow = multiselect.options[0].isSelectedFlow(state)
+            .onEach { flowOutput.send(it) }
+            .launchIn(this)
+        assertFalse("Initially not selected") { flowOutput.receive(100) }
+
+        multiselect.options[0].toggleSelected(state)
+        println("Toggled this option to true")
+        assertTrue(flowOutput.receive(100))
```

Comment:
I noticed ```100``` being used a lot, is that there a specific reason, or is it just a consistent number to use?

Reply:
just a consistent number of milliseconds for the timeout. The value should be immediately pushed to the `flowOutput` channel, but because of threading there could be a minimal delay.

We want to wait for this delay, but not wait indefinitely.

Reply:
Once this [PR](https://github.com/Kotlin/kotlinx.coroutines/issues/1996) is implemented in coroutines this logic will be more straightforward because the `runBlockingTest` method provided by coroutines-test is more predictable in how multiple coroutines execute.